windows/test-app.ts

```diff
@@ -0,0 +1,300 @@
+
+        fs.writeFileSync(fullDestPath, content, {
+            encoding: 'utf8',
+            mode: srcPermissions,
+        });
+    }
+}
+
+/**
+ * Same as 'cp' on Unix. Don't do any replacements.
+ */
+function copyBinaryFile(
+    srcPath: string,
+    destPath: string,
+    cb: (err?: Error) => void,
+) {
+    let cbCalled = false;
+    const srcPermissions = fs.statSync(srcPath).mode;
+    const readStream = fs.createReadStream(srcPath);
+    readStream.on('error', err => {
+        done(err);
+    });
+    const writeStream = fs.createWriteStream(destPath, {
+        mode: srcPermissions,
+    });
+    writeStream.on('error', err => {
+        done(err);
+    });
+    writeStream.on('close', () => {
+        done();
+    });
+    readStream.pipe(writeStream);
+    function done(err?: Error) {
+        if (!cbCalled) {
+            cb(err);
+            cbCalled = true;
+        }
+    }
+}
+
+export function createDir(destPath: string) {
+    if (!fs.existsSync(destPath)) {
+        fs.mkdirSync(destPath);
+    }
+}
+
+export async function copyAndReplaceAll(
+    srcPath: string,
+    destPath: string,
+    relativeDestDir: string,
+    replacements: Replacements = {}
+) {
+    for (const absoluteSrcFilePath of walk(srcPath)) {
+        const filename = path.relative(srcPath, absoluteSrcFilePath);
+        const relativeDestPath = path.join(relativeDestDir, filename);
+        await copyAndReplace(
+            absoluteSrcFilePath,
+            destPath,
+            relativeDestPath,
+            replacements
+        );
+    }
+}
+
+const cppSourceFilesExtensions = ['.h', '.cpp', '.idl', '.xaml'];
+
+export async function copySourceFiles(
+    srcPath: string,
+    destPath: string,
+    relativeDestDir: string
+) {
+    for (const absoluteSrcFilePath of fs.readdirSync(srcPath).map(_ => path.join(srcPath, _))) {
+        if (cppSourceFilesExtensions.indexOf(path.extname(absoluteSrcFilePath)) != -1) {
+            const filename = path.relative(srcPath, absoluteSrcFilePath);
+            const relativeDestPath = path.join(relativeDestDir, filename);
+
+            await copyAndReplace(
+                absoluteSrcFilePath,
+                destPath,
+                relativeDestPath
+            );
+        }
+    }
+}
+
+export async function copyProjectTemplateAndReplace() {
+    if (!srcRootPath) {
+        throw new Error('Need a path to copy from');
+    }
+
+    if (!destPath) {
+        throw new Error('Need a path to copy to');
+    }
+
+    if (!reactNativeModulePath) {
+        throw new Error('react-native-windows node module is not installed');
+    }
+
+    if (!testAppNodeModulePath) {
+        throw new Error('react-native-test-app node module is not installed');
+    }
+
+    const projDir = 'ReactTestApp';
+
+    createDir(path.join(destPath, windowsDir));
+    createDir(path.join(destPath, windowsDir, projDir));
+    createDir(path.join(destPath, windowsDir, projDir, 'Bundle'));
```

Comment:
We should use [`fs.mkdirSync`](https://nodejs.org/api/fs.html#fs_fs_mkdirsync_path_options) here:
```suggestion
    fs.mkdirSync(path.join(destPath, windowsDir, projDir, 'Bundle'), { 'recursive': true });
```